### PR TITLE
Enforce DefaultFunctionTimeout for LinuxConsumption

### LINKED ISSUE #5902: Default consumption sku function timeout should apply to Linux also
Both windows and linux consumption apps should have the same default function timeouts.

https://github.com/Azure/azure-functions-host/blob/ded932ba6de4353bf466e0afb6732d4dba208e5b/src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs#L73

### PR DESCRIPTION
**Pull request by pragnagopa**
### Issue describing the changes in this PR

resolves #5902 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs [hunk 1]
```diff
@@ -156,6 +156,17 @@ public static bool IsWindowsConsumption(this IEnvironment environment)
             return string.Equals(value, ScriptConstants.DynamicSku, StringComparison.OrdinalIgnoreCase);
         }
 
+        /// <summary>
+        /// Gets a value indicating whether the application is running in a Windows or Linux Consumption (dynamic)
+        /// App Service environment.
+        /// </summary>
+        /// <param name="environment">The environment to verify</param>
+        /// <returns><see cref="true"/> if running in a Windows or Linux Consumption App Service app; otherwise, false.</returns>
+        public static bool IsConsumption(this IEnvironment environment)
```
Comment (balag0) [inline src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: Rename to IsConsumptionSku? ( that is how this method is named in dev branch)

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs [hunk 2]
```diff
@@ -78,9 +78,9 @@ private void ConfigureFunctionTimeout(ScriptJobHostOptions options)
         {
             if (options.FunctionTimeout == null)
             {
-                options.FunctionTimeout = _environment.IsWindowsConsumption() ? DefaultFunctionTimeoutDynamic : DefaultFunctionTimeout;
+                options.FunctionTimeout = _environment.IsConsumption() ? DefaultFunctionTimeoutDynamic : DefaultFunctionTimeout;
             }
-            else if (!_environment.IsWindowsConsumption() && TimeSpan.Compare(options.FunctionTimeout.Value, TimeSpan.FromDays(-1)) == 0)
+            else if (!_environment.IsConsumption() && TimeSpan.Compare(options.FunctionTimeout.Value, TimeSpan.FromDays(-1)) == 0)
```
Comment (balag0) [inline src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: One more place
https://github.com/Azure/azure-functions-host/blob/916c255778588951371ffc0da74b7a2c3f1ec2b8/src/WebJobs.Script/Config/ScriptJobHostOptionsSetup.cs#L100

File: src/WebJobs.Script/Environment/EnvironmentExtensions.cs (hunk 1 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Environment/EnvironmentExtensions.cs]: done!

File: src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs (hunk 2 above)
Comment (pragnagopa) [reply src/WebJobs.Script/Config/ScriptHostOptionsSetup.cs]: thanks! addressed!

**Review by balag0 — APPROVED** [review]

**Pull request merged**